php-frontend/src/main/java/org/sonar/php/parser/NewPHPGrammar.java

```diff
@@ -0,0 +1,1498 @@
+        b.token(ELIPSIS),
+        EXPRESSION()
+      ));
+  }
+
+  public FunctionCallTree FUNCTION_CALL_ARGUMENT_LIST() {
+    return b.<FunctionCallTree>nonterminal(PHPLexicalGrammar.FUNCTION_CALL_PARAMETER_LIST)
+      .is(f.functionCallParameterList(
+        b.token(LPARENTHESIS),
+        b.optional(f.newTuple9(
+          FUNCTION_CALL_ARGUMENT(), b.zeroOrMore(f.newTuple5(b.token(COMMA), FUNCTION_CALL_ARGUMENT())))),
+        b.token(RPARENTHESIS)));
+  }
+
+  public ExpressionTree FUNCTION_CALL_ARGUMENT() {
+    return b.<ExpressionTree>nonterminal()
+      .is(
+        b.firstOf(
+          REFERENCE_VARIABLE(),
+          SPREAD_ARGUMENT(),
+          EXPRESSION(),
+          YIELD_EXPRESSION()));
+  }
+
+  public ExpressionTree MEMBER_EXPRESSION() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.MEMBER_EXPRESSION)
+      .is(f.memberExpression(
+        PRIMARY_EXPRESSION(),
+        b.zeroOrMore(
+          b.firstOf(
+            OBJECT_MEMBER_ACCESS(),
+            CLASS_MEMBER_ACCESS(),
+            DIMENSIONAL_OFFSET(),
+            FUNCTION_CALL_ARGUMENT_LIST()))));
+  }
+
+  public MemberAccessTree OBJECT_MEMBER_ACCESS() {
+    return b.<MemberAccessTree>nonterminal(PHPLexicalGrammar.OBJECT_MEMBER_ACCESS)
+      .is(f.objectMemberAccess(
+        b.token(ARROW),
+        b.firstOf(
+          VARIABLE_WITHOUT_OBJECTS(),
+          OBJECT_DIMENSIONAL_LIST(),
+          IDENTIFIER())));
+  }
+
+  public ExpressionTree OBJECT_DIMENSIONAL_LIST() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.OBJECT_DIM_LIST)
+      .is(f.objectDimensionalList(
+        b.firstOf(
+          IDENTIFIER(),
+          f.variableName(b.token(PHPLexicalGrammar.KEYWORDS)),
+          COMPUTED_VARIABLE_NAME()),
+        b.zeroOrMore(
+          b.firstOf(
+            ALTERNATIVE_DIMENSIONAL_OFFSET(),
+            DIMENSIONAL_OFFSET()))));
+  }
+
+  public MemberAccessTree CLASS_MEMBER_ACCESS() {
+    return b.<MemberAccessTree>nonterminal(PHPLexicalGrammar.CLASS_MEMBER_ACCESS)
+      .is(f.classMemberAccess(
+        b.token(DOUBLECOLON),
+        b.firstOf(
+          VARIABLE_WITHOUT_OBJECTS(),
+          IDENTIFIER(),
+          b.token(CLASS),
+          COMPUTED_VARIABLE_NAME())));
+  }
+
+  public LexicalVariablesTree LEXICAL_VARIABLES() {
+    return b.<LexicalVariablesTree>nonterminal(Kind.LEXICAL_VARIABLES)
+      .is(f.lexicalVariables(
+        b.token(USE),
+        b.token(LPARENTHESIS),
+        LEXICAL_VARIABLE(),
+        b.zeroOrMore(f.newTuple11(b.token(COMMA), LEXICAL_VARIABLE())),
+        b.token(RPARENTHESIS)
+      ));
+  }
+
+  public VariableTree LEXICAL_VARIABLE() {
+    return b.<VariableTree>nonterminal(PHPLexicalGrammar.LEXICAL_VAR)
+      .is(f.lexicalVariable(
+        b.optional(b.token(AMPERSAND)),
+        VARIABLE_IDENTIFIER()
+      ));
+  }
+
+  public ExpressionTree STATIC_SCALAR() {
+    // FIXME (PY) : In the old grammar: b.firstOf(COMBINED_SCALAR, CONDITIONAL_EXPR)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1134)